llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -868,6 +878,9 @@ static bool isNoWrap(PredicatedScalarEvolution &PSE, const SCEVAddRecExpr *AR,
       return true;
   }
 
+  if (!Ptr)
+    return false;
+
```

Comment:
Hmpf, if Ptr is nullptr, Assume is ignored: perhaps it makes sense to add a comment about this in the function's doc?

Reply:
If AR is stripped from isNoWrapAddRec (actually isNoWrapGEP), the check for it in this function can be moved after the Ptr check.